pkg/execution/executor/executor.go

```diff
@@ -603,6 +603,9 @@ func (e *executor) Schedule(ctx context.Context, req execution.ScheduleRequest)
 				}
 			}
 			singletonConfig = &queue.Singleton{Key: singletonKey}
+		case errors.Is(err, singleton.ErrEvaluatingSingletonExpression):
+			// Ignore singleton expressions if we cannot evaluate them
+			logger.StdlibLogger(ctx).Warn("error evaluating singleton expression", "error", err)
```

Comment:
Should we treat invalid expressions as a specific key so that singletons still work?  I'm wondering what's preferable:

- Having many more functions run than expected
- Or having far fewer runs than expected.

I also suspect we should shoot over an Inngest event here / an audit trail event so that we can alert and email on this in the future or something.  We need to define how to show alerts / warnings like this in our UI at some point... badly!

Reply:
Or a different specific key and document that